k2/csrc/properties.cc

```diff
@@ -50,4 +86,28 @@ bool IsEpsilonFree(const Fsa &fsa) {
   return true;
 }
 
+bool IsConnected(const Fsa &fsa) {
+  StateId num_states = fsa.NumStates();
```

Comment:
We have to check whether the input fsa is empty.

Reply:
will refactor with `ConnectCore`